api/src/main/java/com/spotify/ffwd/util/HighFrequencyDetector.java

```diff
@@ -123,15 +128,23 @@ public HighFrequencyDetector() {
 
   private int computeTimeDelta(List<Metric> list) {
     int size = list.size();
-    return IntStream.range(1, size)
+    IntSummaryStatistics stats = IntStream.range(1, size)
         .map(
             x ->
                 (int)
                     (list.get(size - x).getTime().getTime()
                         - list.get(size - x - 1).getTime().getTime()))
         .filter(d -> (d >= 0 && d < minFrequencyMillisAllowed))
-        .min()
-        .orElse(-1);
+        .summaryStatistics();
+
+    int result = -1;
+
+    if (stats.getCount() > 5) {
```

Comment:
What is the rational of using 5 as the threshold?  It is better not to hardcode the threshold. 
Public static final int THRESHOLD = 5;  if(stats.getCount() > THRESHOLD)

Reply:
It's magic number :) - I'll change

Reply:
:), I am curious why 5

Reply:
It is based on preliminary testing across 65 canary instances.